feature/auth/src/androidUnitTest/kotlin/org/mifospay/feature/auth/login/LoginViewModelTest.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2025 Mifos Initiative
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * See https://github.com/openMF/mobile-wallet/blob/master/LICENSE.md
+ */
+package org.mifospay.feature.auth.login
+
+import androidx.lifecycle.SavedStateHandle
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+import org.mifospay.core.common.DataState
+import org.mifospay.core.domain.LoginUseCase
+import org.mifospay.core.model.user.UserInfo
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertIs
+import kotlin.test.assertTrue
+
+class MainDispatcherRule : TestWatcher() {
+    val testDispatcher = StandardTestDispatcher()
+
+    override fun starting(description: Description) {
+        Dispatchers.setMain(testDispatcher)
+    }
+
+    override fun finished(description: Description) {
+        Dispatchers.resetMain()
+    }
+}
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class LoginViewModelTest {
+    @get:Rule
+    val mainDispatcherRule = MainDispatcherRule()
+
+    private lateinit var loginUseCase: LoginUseCase
+    private lateinit var savedStateHandle: SavedStateHandle
+    private lateinit var viewModel: LoginViewModel
+
+    @Before
+    fun setup() {
+        loginUseCase = mockk()
+        savedStateHandle = SavedStateHandle()
+        viewModel = LoginViewModel(
+            loginUseCase = loginUseCase,
+            savedStateHandle = savedStateHandle,
+        )
+    }
+
+    @Test
+    fun `initial state has empty username and password`() = runTest {
+        assertEquals("", viewModel.state.username)
+        assertEquals("", viewModel.state.password)
+        assertFalse(viewModel.state.isPasswordVisible)
+        assertEquals(null, viewModel.state.dialogState)
+    }
+
+    @Test
+    fun `username change updates state`() = runTest {
```

Comment:
- The naming convention for some of your unit tests doesn't show what it achieve the result is vague. I personally use`thingUnderTest_trigger_result` naming convention used for testing viewmodel from `android developers`. link here (https://developer.android.com/codelabs/basic-android-kotlin-compose-test-viewmodel?continue=https%3A%2F%2Fdeveloper.android.com%2Fcourses%2Fpathways%2Fandroid-basics-compose-unit-4-pathway-1%23codelab-https%3A%2F%2Fdeveloper.android.com%2Fcodelabs%2Fbasic-android-kotlin-compose-test-viewmodel#3)
- You can ask @therajanmaurya or @niyajali for naming convention

<img width="797" alt="Screenshot 2025-03-27 at 02 08 07" src="https://github.com/user-attachments/assets/58e21eb1-77a7-46bd-b85e-54cd1107694d" />